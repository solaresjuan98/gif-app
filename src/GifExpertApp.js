import React, { useState } from 'react'
import { AddCategory } from './compontents/AddCategory';
import { GifGrid } from './compontents/GifGrid';

export const GifExpertApp = () => {

    /* const catergories = ['Nintendo', 'Xbox', 'PlayStation'];// one punch, samurai x, dragon ball */
    const [categories, setCategories] = useState(['Nintendo']);

    /*     const handleAdd = () => {
    
            //setCategories([...categories, 'Phillips']);
            setCategories(categories => [...categories, 'Phillips']);
        } */

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories} />
            <hr />

            <br />

            <ol>
                {
                    categories.map(category => (
                        <GifGrid
                            key={category}
                            category={category}
                        />
                    ))
                }
            </ol>
        </>
    )
}
