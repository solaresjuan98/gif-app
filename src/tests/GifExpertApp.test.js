import React from 'react';
import { shallow } from 'enzyme';
import {GifExpertApp} from '../GifExpertApp';

describe('tests on <GifExpertApp />', () => {
   
    
    test('should to show component correctly', () => {
        const wrapper = shallow(<GifExpertApp category={'Nintendo'} />);
        expect(wrapper).toMatchSnapshot();
    });
    
    test('should show a category list', () => {
        
        const categories = ['Nintendo', 'PlayStation'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    
});
