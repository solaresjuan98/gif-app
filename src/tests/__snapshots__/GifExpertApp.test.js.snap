// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <GifExpertApp /> should show a category list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <br />
  <ol>
    <GifGrid
      category="Nintendo"
      key="Nintendo"
    />
    <GifGrid
      category="PlayStation"
      key="PlayStation"
    />
  </ol>
</Fragment>
`;

exports[`tests on <GifExpertApp /> should to show component correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <br />
  <ol />
</Fragment>
`;
